### PR TITLE
5814 duplicate affected party records per transaction

### PR DESCRIPTION
**Pull request by daveb-hni**
Updated auditing on all transactions to include a value for the new column 'direction' on AffectedParty.

### CODE REVIEW

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/GroupMemberController.java [hunk 1]
```diff
@@ -82,7 +83,7 @@
 			logger.info("addDependentResponse response: {} ", addDependentResponse);
 			
 			transactionComplete(transaction);
-			addAffectedParty(transaction, IdentifierType.PHN, addDependentResponse.getPhn());
+			addAffectedParty(transaction, IdentifierType.PHN, addDependentResponse.getPhn(), AffectedPartyDirection.INBOUND);
```
Comment (AnumehaSrivastava05) [inline backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/GroupMemberController.java:86]: Should it be INBOUND or OUTBOUND? What if dependent has already been added. There shouldn't be any entry in that scenario?

File: backend/src/test/java/ca/bc/gov/hlth/hnweb/BaseControllerTest.java [hunk 2]
```diff
@@ -73,4 +82,22 @@ protected Transaction assertTransactionCreated(TransactionType type) {
 		return opt.get();
 	}
 	
+	/**
+	 * Asserts the the Transaction is created.
```
Comment (AnumehaSrivastava05) [inline backend/src/test/java/ca/bc/gov/hlth/hnweb/BaseControllerTest.java:86]: Seems typo. Remove extra the

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/GroupMemberController.java (hunk 1 above)
Comment (daveb-hni) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/GroupMemberController.java:86]: All PHNs are logged for both Outbound and Inbound. The direction makes the entry unique. So the same PHN can logged for both directions if it's in the request and the response. This is logging the Inbound/response PHN from addDependentResponse.getPhn().

File: backend/src/test/java/ca/bc/gov/hlth/hnweb/BaseControllerTest.java (hunk 2 above)
Comment (daveb-hni) [reply backend/src/test/java/ca/bc/gov/hlth/hnweb/BaseControllerTest.java:86]: The entire comment was incorrect so updated it all, fixed typo in assertTransactionCreated method description

**Review by AnumehaSrivastava05 — APPROVED** [review]

**Pull request merged**